### PR TITLE
@metamask/controllers@15.0.0

### PR DESCRIPTION
**Pull request by rekmarks**
Adds the latest version of `@metamask/controllers`, and updates our usage of the `ApprovalController`, which has been migrated to `BaseControllerV2`. Of [the new `controllers` release](https://github.com/MetaMask/controllers/releases/tag/v15.0.0), only the `ApprovalController` migration should be breaking.

This is the first time we use events on the `ControllerMessenger` to update the badge, so I turned the messenger into a property on the main `MetaMaskController` in order to subscribe to events on it in `background.js`. I confirmed that the badge does indeed update during local QA.

As it turns out, [MetaMask/controllers#571](https://github.com/MetaMask/controllers/pull/571) was breaking for a single unit test case, which is now handled during setup and teardown for the related test suite (`metamask-controller.test.js`).

### CODE REVIEW

File: app/scripts/metamask-controller.test.js [hunk 1]
```diff
@@ -109,6 +109,7 @@ describe('MetaMaskController', function () {
   const noop = () => undefined;
 
   before(async function () {
+    globalThis.AbortController = window.AbortController;
```
Comment (rekmarks) [inline app/scripts/metamask-controller.test.js:112]: As an aside, that I had to do this has me wondering when we'll fully migrate to `jest`.

File: app/scripts/background.js [hunk 2]
```diff
@@ -435,12 +435,16 @@ function setupController(initState, initLangCode) {
...
-  controller.approvalController.subscribe(updateBadge);
   controller.appStateController.on(
     METAMASK_CONTROLLER_EVENTS.UPDATE_BADGE,
     updateBadge,
   );
 
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.APPROVAL_STATE_CHANGE,
+    updateBadge,
+  );
```
Comment (shanejonas) [inline app/scripts/background.js:446]: somewhat related nit: we never clean up any of these listeners in this file. 

this main controller only gets used once in the lifecycle so we dont need to, but just wanted to point it out

Comment (rekmarks) [reply app/scripts/background.js:446]: I'm pretty confident that we do have event-related leaks, so it's great to keep an eye out for it!

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]
LGTM! Most of the breaking changes in this release don't have any impact it seems. Just the ApprovalController really.

**Review by shanejonas — APPROVED** [review]

**Pull request merged**